source/NationalInstruments/jquery.flot.historybuffer.analogWaveform.js

```diff
@@ -158,20 +169,38 @@ console.log(hb1.toDataSeries()); //[[4, 1], [5, 2], [6, 3], [null, null], [1, 1]
         var result = [];
         var waveforms = this.buffers[index].toArray();
         var previousWaveform = waveforms[0];
+        var beforeSkippedWaveformIndex = -1;
+        var afterSkippedWaveformIndex = -1;
+        var i = -1;
 
         waveforms.forEach(function (waveform) {
-            if (!waveformInRange(waveform, start, end)) {
-                return;
+            i++;
```

Comment:
The second param in the `forEach` callback is the current index, you could use that instead of `i`.

Reply:
Done.